runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java

```diff
@@ -105,7 +105,14 @@ public DataflowUnboundedReadFromBoundedSource(BoundedSource<T> source) {
 
   @Override
   public String getKindString() {
-    return "Read(" + approximateSimpleName(source.getClass()) + ")";
+    String sourceName;
+    if (source.getClass().isAnonymousClass()) {
+      sourceName = "AnonymousSource";
```

Comment:
Why not include the anonymous class name? That would at least tell us which package/class defined it?